### DISCUSSION TITLE
Uptime.com-authentication issue

### DISCUSSION THREAD

**Discussion by alandreasyans** [grafana/grafana-infinity-datasource#1037] [Specific APIs]
## **Describe the bug**

I want to connect to uptime.com to be able to get the status of my services and visualize in the grafana, however getting an issue error getting data frame. 401 Unauthorized. As for the prefix it uses not a bearer token I've used the API key value pair type for the authentication and specified "Token" as the key and the token itself as value.
With a usual curl command I was able to get details from uptime.com

Below is the official documentation of uptime.com how the token can be used.

API Access
To access our API programatically, you must pass one of the below API keys via a HTTP Header with a key name of "Authorization" and a key prefix of "Token" with the below API Key.

Raw HTTP Example:

GET /api/v1/
Authorization: Token token_value
CURL Example:

curl -H "Authorization: Token token_value" https://uptime.com/api/v1/

## **Steps to reproduce**

Provide steps to reproduce the issue.

Register for uptime.com(it has a trial option) then by using a token try to visualize some data with it by providing the https://uptime.com/api/v1/ as the URL and authentication mechanism as described.

## **Screenshots**

![image](https://github.com/grafana/grafana-infinity-datasource/assets/112861637/e8359220-fb41-4952-8eb5-21bb226cc4de)

## **Version Details:**

- Grafana version : Grafana v10.4.1
- Plugin version : 2.7.0 and 2.7.1

## **Additional context**

Thanks in advance!!

**Comment by wezlu** [comment]
The `Authorization: Token token_value` is missing in the query options of Grafana data source based from the shared screenshot. This is likely the cause of the 401 unauthorized error result.

Configuring/adding the [API key authentication](https://grafana.com/docs/plugins/yesoreyeram-infinity-datasource/latest/setup/authentication/#api-key) settings in the Grafana query options would solve this:
```
Key:   Token
Value: <token_value>
In:    header
```
